File: java/L000/LeetCode0061.java
package L000;

import util.ListNode;

public class LeetCode0061 {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null) {
            return head;
        }
        int n = 0;
        ListNode t = head;
        while (t != null) {
            ++n;
            t = t.next;
        }
        k = k % n;
        if (k == 0) {
            return head;
        }
        t = head;
        for (int i = 0;i < n - k - 1;++i) {
            t = t.next;
        }
        ListNode nh = t.next;
        t.next = null;
        t = nh;
        while (t.next != null) {
            t = t.next;
        }
        t.next = head;
        return nh;
    }
}

File: cpp/L000/LeetCode0061.cc
#include "../heads/list_node.h"

class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (head == nullptr) {
            return nullptr;
        }
        ListNode fisrt(0);
        fisrt.next = head;
        ListNode *tail = &fisrt;
        int len = 0;
        while (tail->next != nullptr) {
            ++len;
            tail = tail->next;
        }
        ListNode *hp = &fisrt;
        k = k % len;
        for (int i = 0;i < len - k;++i) {
            hp = hp->next;
        }
        tail->next = head;
        ListNode *n = hp->next;
        hp->next = nullptr;
        return n;
    }
};
